db/versions/0111.yml

```diff
@@ -0,0 +1,65 @@
+version: 111
+description: fetch audit history for a client
+migrationScript: |-
+  begin
+    CREATE INDEX audit_history_client_idx ON audit_history(client_id);
+  end
+downgradeScript: |-
+  begin
+    DROP INDEX if exists audit_history_client_idx;
+  end
+methods:
+  get_audit_history:
+    deprecated: true
+  get_combined_audit_history:
+    description: |
+      Get Audit History by client ID OR by entity ID and entity type.
+      If client_id_in is provided, returns full audit records for that client.
+      If entity_id_in and entity_type_in are provided, returns filtered audit records for that entity.
+      At least one search parameter must be provided.
+    mode: read
+    serviceName: auth
+    args: client_id_in text, entity_id_in text, entity_type_in text, page_size_in integer, page_offset_in integer
+    returns:
+      table (
+        entity_id text,
+        entity_type text,
+        client_id text,
+        action_type text,
+        created timestamptz
+      )
+    body: |-
+      begin
+        -- Handle case where no search criteria is provided
+        if client_id_in is null and (entity_id_in is null or entity_type_in is null) then
+          raise exception 'At least client_id or both entity_id and entity_type must be provided';
+        end if;
+        if client_id_in is not null and (entity_id_in is null or entity_type_in is null) then
```

Comment:
I'm wondering if we should make it that complicated here. 
It makes sense to check if entity type is provided when entity id is in, but otherwise we can allow filter by both at the same time, why not.

we can use similar approach like in https://github.com/taskcluster/taskcluster/blob/511974dacf86b36e8096f39498da1cbeedbfe7a8/db/versions/0105.yml#L427-L439

```sql
(audit_history.client_id = client_id_in OR client_id_in IS NULL) AND
(
   (audit_history.entity_id = entity_id_in AND audit_history.entity_type = entity_type_in) OR entity_id_in IS NULL
)
```
(needs checking ^ :)